app/src/pages/project/SessionsTable.tsx

```diff
@@ -0,0 +1,305 @@
+/* eslint-disable react/prop-types */
+import React, {
+  startTransition,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { graphql, usePaginationFragment } from "react-relay";
+import { useNavigate } from "react-router";
+import {
+  ColumnDef,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from "@tanstack/react-table";
+import { css } from "@emotion/react";
+
+import { Flex, Icon, Icons, View } from "@arizeai/components";
+
+import { selectableTableCSS } from "@phoenix/components/table/styles";
+import { TimestampCell } from "@phoenix/components/table/TimestampCell";
+import { useStreamState } from "@phoenix/contexts/StreamStateContext";
+import { useTracingContext } from "@phoenix/contexts/TracingContext";
+
+import { IntCell, TextCell } from "../../components/table";
+import { TokenCount } from "../../components/trace/TokenCount";
+
+import { SessionsTable_sessions$key } from "./__generated__/SessionsTable_sessions.graphql";
+import { SessionsTableQuery } from "./__generated__/SessionsTableQuery.graphql";
+import { ProjectTableEmpty } from "./ProjectTableEmpty";
+import { SpanFilterConditionField } from "./SpanFilterConditionField";
+import { spansTableCSS } from "./styles";
+
+type SessionsTableProps = {
+  project: SessionsTable_sessions$key;
+};
+
+const PAGE_SIZE = 50;
+
+export function SessionsTable(props: SessionsTableProps) {
+  //we need a reference to the scrolling element for logic down below
+  const tableContainerRef = useRef<HTMLDivElement>(null);
```

Comment:
is this for pagination? if so would add that to this comment or in either case remove it all together

Reply:
the comment now mentions pagination